webtasks/cron_mongodb.js

```diff
@@ -0,0 +1,230 @@
+                res.writeHead(200, {'Content-Type': 'application/json'});
+                res.end(JSON.stringify(data));
+            })
+            .catch(function (err) {
+                error(err.statusCode || 500, err);
+            });
+    } else if (action === 'list_jobs') {
+        // List all scheduled webtasks or only those in a container passed via query
+        
+        return withMongoCollection(context.data.MONGO_COLLECTION)
+            .then(function (coll) {
+                var query = {};
+                
+                // Both admin and user will hit this endpoint. When admin, container can be unset
+                if (context.data.container) query.container = context.data.container;
+                
+                // TODO (ggoodman): Pagination logic
+                return coll.findAsync(query)
+                    .then(function (cursor) {
+                        return Bluebird.promisify(cursor.toArray, cursor)();
+                    });
+            })
+            .then(function (data) {
+                res.writeHead(200, {'Content-Type': 'application/json'});
+                res.end(JSON.stringify(data));
+            })
+            .catch(function (err) {
+                error(err.statusCode || 500, err);
+            });
+    
+    } else if (action === 'get_job') {
+        // Get a single scheduled webtask
+        
+        if (!validate_params(['container', 'name']))
+            return;
+            
+        return withMongoCollection(context.data.MONGO_COLLECTION)
+            .then(function (coll) {
+                var query = {
+                    container: context.data.container,
+                    name: context.data.name,
+                };
+                
+                return coll.findOneAsync(query, {});
+            })
+            .then(function (data) {
+                res.writeHead(200, {'Content-Type': 'application/json'});
+                res.end(JSON.stringify(data));
+            })
+            .catch(function (err) {
+                error(err.statusCode || 500, err);
+            });
+    
+    } else if (action === 'destroy_job') {
+        // Get a single scheduled webtask
+        
+        if (!validate_params(['container', 'name']))
+            return;
+            
+        return withMongoCollection(context.data.MONGO_COLLECTION)
+            .then(function (coll) {
+                var filter = {
+                    container: context.data.container,
+                    name: context.data.name,
+                };
+                
+                return coll.deleteOne(filter, {w: 1});
+            })
+            .then(function (data) {
+                res.writeHead(204);
+                res.end();
+            })
+            .catch(function (err) {
+                error(err.statusCode || 500, err);
+            });
+    
+    } else {
+        res.writeHead(404);
+        res.end('Method not found');
+    }
+    
```

Comment:
I understand there is no renew and release yet. One thing to watch for that just popped to my mind: these need to be designed to be resistant to race conditions with destroy. If task was destroyed, I think releasing should be a no-op and succeed while renewing should fail.